pkg/nsx/services/subnetport/subnetport.go

```diff
@@ -95,66 +96,74 @@ func (service *SubnetPortService) CreateOrUpdateSubnetPort(obj *v1alpha1.SubnetP
 		err = service.NSXClient.PortClient.Patch(subnetInfo.OrgID, subnetInfo.ProjectID, subnetInfo.VPCID, subnetInfo.ID, *nsxSubnetPort.Id, *nsxSubnetPort)
 		if err != nil {
 			log.Error(err, "failed to create or update subnet port")
-			return err
+			return nil, err
 		}
 		err = service.SubnetPortStore.Operate(nsxSubnetPort)
 		if err != nil {
-			return err
+			return nil, err
 		}
 		if existingSubnetPort != nil {
 			log.Info("updated NSX subnet port", "subnetPort", nsxSubnetPort)
 		} else {
 			log.Info("created NSX subnet port", "subnetPort", nsxSubnetPort)
 		}
 	}
-	if err := service.CheckSubnetPortState(obj, nsxSubnetPath); err != nil {
-		log.Error(err, "check and update NSX subnet port state failed, would retry exponentially", "subnetport", obj.UID)
-		return err
+	nsxSubnetPortState, err := service.CheckSubnetPortState(obj, nsxSubnetPath)
```

Comment:
could we just pass "obj" as param in this func? when checking subnet port realization state, we can get NSX Subnet Port path using ID from store, then parse the orgID/project ID/VPC ID etc.
It is strange that we pass "nsxSubnetPath" here.

Reply:
The parameter `nsxSubnetPath` can't be removed.... Because subnetport controller need it to create NSX subnetport, the value is provided by service mediator, for now, the service mediator only can be invoked from the controller level.